src/opm/output/eclipse/InteHEAD.cpp

```diff
@@ -910,14 +970,47 @@ namespace {
         return maxNumActiveConn;
     }
 
+    template <typename AquiferCollection>
+    int maxAquID(const AquiferCollection& aquiferCollection)
+    {
+        return std::accumulate(aquiferCollection.begin(), aquiferCollection.end(), 0,
+                               [](const int maxID, const auto& aquiferData)
+                               {
+                                   return std::max(maxID, aquiferData.aquiferID);
+                               });
+    }
+
+    template <typename AquFluxIter>
+    int maxAquID(AquFluxIter begin, AquFluxIter end)
+    {
+        auto maxIDPos =
+            std::max_element(begin, end,
+                             [](const auto& aq1, const auto& aq2)
+                             {
+                                 return aq1.first < aq2.first;
+                             });
+
+        return (maxIDPos == end) ? 0 : maxIDPos->first;
+    }
+
     int getMaximumAnalyticAquiferID(const Opm::AquiferConfig& cfg)
     {
-        const auto& aquifer_ids = analyticAquiferIDs(cfg);
-        if (!aquifer_ids.empty()) {
-            return *max_element(std::begin(aquifer_ids), std::end(aquifer_ids));
-        } else {
-            return 0;
-        }
+        return std::max({
```

Comment:
There you go. Didn't even know this one. Cool to learn.

Reply:
Yeah, `max(initializer_list<T>)` is incredibly useful for these cases.